scripts/functions.R

```diff
@@ -0,0 +1,56 @@
+#Birth Rates
+
+total <- 'Total births'
+age <- c('15 and younger ', '15-19 ', '20-24 ', '25-29 ', '30-34 ', '35-39 ', '40-45 ', '45 or older ')
+race <- c('White ', 'Black ', 'Asian or Pacific Islander ', 'American Indian or Alaska Native ')
+ethnicity <- c('Hispanic ', 'Non-Hispanic ', 'Non-Hispanic White ', 'Non-Hispanic Black ')
+marrital_status <- c('Married ', 'Unmarried ')
+education <- c('Doctorate or professional degree ', 'Master\'s degree ', 'Bachelor\'s degree  ', 'Associate degree ', 'Some college credit, but not a degree ', 'High school, no diploma ', '8th grade or less ', 'High school graduate or GED ')
+
+
+
+birth_graph <- function(group) {
```

Comment:
Generally this is a good function. My biggest criticism is that I typically prefer to generalize a bit further. This function entirely relies upon a data frame called `births_long` existing in your global environment. If that doesn't exist, it would fail.

Here's how I might adjust this function, including the use of direct labeling:

```
birth_graph_direct <- function(df, group) {
  
  group_name <- rlang::as_name(rlang::enquo(group))
  group_name <- str_to_title(group_name)
  group_name <- str_replace_all(group_name, "_", " ")
  
  df_group <-
    df %>% 
    filter(Type %in% group)
  
  df_last <- 
    df_group %>% 
    filter(Year == max(Year))
  
  df_group %>%
    ggplot(aes(x = Year, 
               y = Births, 
               color = Type)) +
    scale_y_continuous(
      labels = scales::comma,
      sec.axis = dup_axis(
        breaks = df_last$Births,
        labels = df_last$Type,
        name = NULL
      )) +
    geom_line(scales = "free_x") +
    labs(title = glue::glue("Number of Births by Mother's {group_name}"),
         y = "") +
    theme_bw() +
    theme(plot.title = element_text(hjust = 0.5),
          legend.position = "none")
  
}
```

The output of your graph would now look something like this:

<img width="692" alt="image" src="https://user-images.githubusercontent.com/16887262/167536762-08a2a7c4-ad34-4250-948f-60b7e5e9d085.png">

The clear challenge here is when there are categories that overlap - for example, "45 or older" and "15 and younger". That can be solved using a package such as [`ggrepel`](https://cran.r-project.org/web/packages/ggrepel/vignettes/ggrepel.html)